ui/page/dcrdex/dcrdex_page.go

```diff
@@ -101,32 +110,38 @@ func (pg *DEXPage) OnNavigatedTo() {
 		}
 
 		pg.dexIsLoading = false
+		pg.showSplashPage = false
 	}()
 }
 
 // Layout draws the page UI components into the provided layout context to be
 // eventually drawn on screen.
 // Part of the load.Page interface.
 func (pg *DEXPage) Layout(gtx C) D {
-	if pg.isDexFirstVisit || pg.dexIsLoading {
+	isMainnet := pg.AssetsManager.NetType() == utils.Mainnet
+	if !isMainnet && (!pg.AssetsManager.DEXCInitialized() || pg.CurrentPage() == nil) { // dexc must have been reset.
+		pg.showSplashPage = true
+		if !pg.dexIsLoading {
+			pg.ParentNavigator().CloseAllPages()
+			pg.prepareInitialPage()
+		}
+	}
```

Comment:
Previously, `!pg.AssetsManager.DEXCInitialized() || pg.CurrentPage() == nil` was treated as dex init error. If there was actually a dex init error, this new code will just keep repeating this block of code. It would mostly be a no-op anyway, but should we show a message to the user if dex init fails?